src/code/z_player_lib.c

```diff
@@ -2035,21 +2599,203 @@ void Player_DrawGetItem(PlayState* play, Player* player) {
+                func_80124618(D_801C05A8, player->skelAnime.curFrame, player->unk_AF0);
+            } else if (player->skelAnime.animation == &gPlayerAnim_pz_swimtowait) {
+                func_80124618(D_801C05D8, player->skelAnime.curFrame, player->unk_AF0);
+            } else if (player->skelAnime.animation == &gPlayerAnim_pz_bladeon) {
+                func_80124618(D_801C07F0, player->skelAnime.curFrame, player->unk_AF0);
+            } else if (player->skelAnime.animation == &gPlayerAnim_pz_jumpAT) {
+                func_80124618(D_801C0820, player->skelAnime.curFrame, player->unk_AF0);
+            } else if (player->skelAnime.animation == &gPlayerAnim_pz_jumpATend) {
+                func_80124618(D_801C0838, player->skelAnime.curFrame, player->unk_AF0);
+            } else if (player->heldItemAction == PLAYER_IA_ZORA_FINS) {
+                player->unk_AF0[0].x = 1.0f;
+                player->unk_AF0[0].y = 1.0f;
+                player->unk_AF0[0].z = 1.0f;
+            } else {
+                player->unk_AF0[0].x = 0.4f;
+                player->unk_AF0[0].y = 0.6f;
+                player->unk_AF0[0].z = 0.7f;
+            }
+            Matrix_Push();
+
+            Matrix_Scale(player->unk_AF0[0].x, player->unk_AF0[0].y, player->unk_AF0[0].z, MTXMODE_APPLY);
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+            gSPDisplayList(POLY_OPA_DISP++, D_801C0AB4[arg2]);
+
+            if (player->meleeWeaponState != 0) {
+                if ((((player->meleeWeaponAnimation == PLAYER_MWA_ZORA_PUNCH_LEFT)) && (arg2 == 0)) ||
+                    ((player->meleeWeaponAnimation == PLAYER_MWA_ZORA_PUNCH_COMBO) && (arg2 != 0))) {
+                    func_8012669C(play, player, D_801C0A00, D_801C09DC);
+                }
+            }
+
+            Matrix_Pop();
+        }
+
+        if (player->skelAnime.animation == &gPlayerAnim_pz_waterroll) {
+            func_80124618(D_801C05C8, player->skelAnime.curFrame, player->unk_AF0);
+            D_801C05F0[0].unk_2.x = 0x32;
+            D_801C05F0[0].unk_2.y = 0x32;
+            D_801C05F0[0].unk_2.z = 0x32;
+            D_801C05F0[1].unk_2 = D_801C05F0[0].unk_2;
+        } else if (player->skelAnime.animation == &gPlayerAnim_pz_swimtowait) {
+            func_80124618(D_801C05F0, player->skelAnime.curFrame, player->unk_AF0);
+        } else if (player->skelAnime.animation == &gPlayerAnim_pz_fishswim) {
+            player->unk_AF0[0].x = (ABS_ALT(player->unk_B8A) * 0.00003f) + 0.5f;
+            player->unk_AF0[0].y = player->unk_AF0[0].x;
+            player->unk_AF0[0].z = player->unk_AF0[0].x;
+
+            D_801C05F0[0].unk_2.x = (player->unk_AF0[0].x * 100.0f);
+
+            D_801C05F0[0].unk_2.y = D_801C05F0[0].unk_2.x;
+            D_801C05F0[0].unk_2.z = D_801C05F0[0].unk_2.x;
+
+            D_801C05F0[1].unk_2 = D_801C05F0[0].unk_2;
+        } else {
+            //! @bug Skips CLOSE_DISPS
```

Comment:
I wonder what the debug version looks like, where it actually does have a function call

Reply:
We looked at this a long time ago and it seems like it is actually missing the function call here
https://discord.com/channels/688807550715560050/688851997352263820/929146776026505306